frontend/src/components/NavBar.js

```diff
@@ -92,6 +92,15 @@ class NavBar extends React.Component<{
 
     createNewWorksheet() {
         this.resetDialog();
+        if (this.state.newWorksheetName.includes(' ')) {
```

Comment:
Can you make this logic match the backend (there's a regex)?  Put a comment to that file to keep it consistent?